src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,201 @@
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int SpiClockFrequency = 5_000_000;
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int DataFlow = 0;
+
+        /// <summary>
+        /// MAX31856 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode1;
+
+        #endregion
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature GetTemperature() => Temperature.FromDegreesCelsius(ReadTemperature());
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature GetCJTemperature() => Temperature.FromDegreesCelsius(ReadCJTemperature());
+
+        /// <summary>
+        /// Creates a new instance of the max31856.
+        /// </summary>
+        /// <param name="spiDevice">The communications channel to a device on a SPI bus</param>
+        /// <param name="thermocoupleType">Thermocouple Type T,K,R</param>
+        public MAX31856(SpiDevice spiDevice, ThermocoupleType thermocoupleType)
+        {
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _thermocoupleType = (byte)thermocoupleType;
+            Initialize();
+        }
+
+        #region private and internal
+
+        /// <summary>
+        /// Standard initialization routine.
+        /// </summary>
+        /// /// <remarks>
+        /// You can add new write lines if you want to alter the settings of the device. Settings can be found in the Technical Manual
+        /// </remarks>
+        private void Initialize()
+        {
+            Span<byte> configurationSetting0 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR0,
+                0x90
+            };
+            Span<byte> configurationSetting1 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR1,
+                _thermocoupleType
+            };
+
+            Write(configurationSetting0);
+            Write(configurationSetting1);
+        }
+
+        /// <summary>
+        /// Command to Get Temperature from the Device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data next it converts the bytes to thermocouple temperature  and throws an exception if there is an error
+        /// </remarks>
+        private double ReadTemperature()
+        {
+            var spiOutputData = WriteRead(Register.READ_CR0, 16);
+            if ((spiOutputData[16] & (byte)Register.ERROR_OPEN) == (byte)Register.ERROR_OPEN)
```

Comment:
So what happen when the SPI device just receive 0x00 on the 17 bytes? You won't then detect an error and the message will be seen as valid. Is there any other way to check that the WriteRead has been successfull?

Reply:
I have added code to read CR0 after initialization of it. If it reads the same value it initializes to then the device is wired correctly. If not then it throws an exception. This is under the new TryGetTemperature class I have a separate exception for the thermocouple being wired properly.